### PR TITLE
Null component update

### PR DESCRIPTION
**Pull request by pandmitr**
Change list:
- Loc:: added;
- component cache added;
- component class added;
- perfect codestyle;
- phpDoc added;
- result_modifier.php added;
- component_epilog.php added;
- php >= 5.4 required.

By pandmitr

### CODE REVIEW

File: local/components/codecraft/null/templates/.default/component_epilog.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+ * @var string                  $templateName
+ * @var string                  $templateFile
+ * @var string                  $templateFolder
+ *
+ * @global CUser                $USER
+ * @global CMain                $APPLICATION
+ * @global CDatabase            $DB
+ */
+
+$APPLICATION->SetTitle($arResult['TITLE']);
```
Comment (Manriel) [inline local/components/codecraft/null/templates/.default/component_epilog.php:23]: If template code returns from cache, $arResult['TITLE'] == NULL, that is incorrect.

Comment (pandmitr) [inline local/components/codecraft/null/templates/.default/component_epilog.php:23]: Yes, it's possible to bx <= 10

Now it problem is not actual, see http://dev.1c-bitrix.ru/learning/course/?COURSE_ID=43&LESSON_ID=2975

Comment (pandmitr) [inline local/components/codecraft/null/templates/.default/component_epilog.php:23]: Sorry, you right. And I has any error, all fixed.

Comment (Manriel) [inline local/components/codecraft/null/templates/.default/component_epilog.php:23]: Yep. :-)
arResult array key must contains in SetResultCacheKeys method parameters to save in cache. This key do not included into cached keys, so, it do not saved in cache.
You can check it.

Comment (pandmitr) [inline local/components/codecraft/null/templates/.default/component_epilog.php:23]: Already fixed, see it

**Pull request merged**